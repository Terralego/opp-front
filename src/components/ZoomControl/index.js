import React from 'react';

import { Slider, AnchorButton } from '@blueprintjs/core';


export class ZoomControl extends React.PureComponent {
  updateScale = eventScale => {
    const { onChange } = this.props;
    onChange(Math.min(Math.max(eventScale, 1), 4));
  }

  decreaseScale = () => {
    const { scale, step } = this.props;
    this.updateScale(scale - step);
  }

  increaseScale = () => {
    const { scale, step } = this.props;
    this.updateScale(scale + step);
  }

  render () {
    const { scale, onChange, step } = this.props;
    return (
      <div className="zoom__compare">
        <AnchorButton
          className="zoom__button"
          icon="zoom-out"
          disabled={scale === 1}
          onClick={this.decreaseScale}
        />
        <div className="zoom__container">
          <Slider
            value={scale}
            min={1}
            max={4}
            stepSize={step}
            onChange={onChange}
          />
        </div>
        <AnchorButton
          className="zoom__button"
          icon="zoom-in"
          disabled={scale === 4}
          onClick={this.increaseScale}
        />
      </div>
    );
  }
}

export default (ZoomControl);
