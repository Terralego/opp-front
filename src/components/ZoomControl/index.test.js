import React from 'react';
import renderer from 'react-test-renderer';

import ZoomControl from './index';

it('should render', () => {
  const onChange = jest.fn();
  const step = 0.5;
  const scale = 1;

  const tree = renderer
    .create(<ZoomControl scale={scale} step={step} onChange={onChange} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
