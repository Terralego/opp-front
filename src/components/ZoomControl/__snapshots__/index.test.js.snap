// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="zoom__compare"
>
  <a
    className="bp3-button bp3-disabled zoom__button"
    disabled={true}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex={-1}
  >
    <span
      className="bp3-icon bp3-icon-zoom-out"
      icon="zoom-out"
    >
      <svg
        data-icon="zoom-out"
        height={16}
        viewBox="0 0 16 16"
        width={16}
      >
        <desc>
          zoom-out
        </desc>
        <path
          d="M3.99 5.99c-.55 0-1 .45-1 1s.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1h-6zm11.56 7.44l-2.67-2.68a6.94 6.94 0 0 0 1.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 1 0 2.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
          fillRule="evenodd"
        />
      </svg>
    </span>
  </a>
  <div
    className="zoom__container"
  >
    <div
      className="bp3-slider"
      onMouseDown={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="bp3-slider-track"
      >
        <div
          className="bp3-slider-progress"
          style={
            Object {
              "left": "0.00%",
              "right": "100.00%",
              "top": 0,
            }
          }
        />
        <div
          className="bp3-slider-progress bp3-intent-primary"
          style={
            Object {
              "left": "0.00%",
              "right": "100.00%",
              "top": 0,
            }
          }
        />
        <div
          className="bp3-slider-progress"
          style={
            Object {
              "left": "0.00%",
              "right": "100.00%",
              "top": 0,
            }
          }
        />
      </div>
      <div
        className="bp3-slider-axis"
      >
        <div
          className="bp3-slider-label"
          style={
            Object {
              "left": "0.00%",
            }
          }
        >
          1.0
        </div>
        <div
          className="bp3-slider-label"
          style={
            Object {
              "left": "0.00%",
            }
          }
        >
          2.0
        </div>
        <div
          className="bp3-slider-label"
          style={
            Object {
              "left": "0.00%",
            }
          }
        >
          3.0
        </div>
        <div
          className="bp3-slider-label"
          style={
            Object {
              "left": "0.00%",
            }
          }
        >
          4.0
        </div>
      </div>
      <span
        className="bp3-slider-handle"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "left": "calc(0.00% - 0px)",
          }
        }
        tabIndex={0}
      >
        <span
          className="bp3-slider-label"
        >
          1.0
        </span>
      </span>
    </div>
  </div>
  <a
    className="bp3-button zoom__button"
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="bp3-icon bp3-icon-zoom-in"
      icon="zoom-in"
    >
      <svg
        data-icon="zoom-in"
        height={16}
        viewBox="0 0 16 16"
        width={16}
      >
        <desc>
          zoom-in
        </desc>
        <path
          d="M7.99 5.99v-2c0-.55-.45-1-1-1s-1 .45-1 1v2h-2c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1v-2h2c.55 0 1-.45 1-1s-.45-1-1-1h-2zm7.56 7.44l-2.67-2.68a6.94 6.94 0 0 0 1.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 1 0 2.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
          fillRule="evenodd"
        />
      </svg>
    </span>
  </a>
</div>
`;
